src/persona/persona.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@nestjs/common';
+import { Model } from 'mongoose';
+import { InjectModel } from '@nestjs/mongoose';
+import { MongoService } from '../mongo-service/mongo.service';
+import { ConnectPersonaInput, connectPersona } from '../shared';
+import {
+  PersonaDocument,
+  PersonaInterface,
+} from './interfaces/persona.interfaces';
+import { PersonaInput } from './input';
+import { PersonaType } from './dto/persona.dto';
+import { UpdatePersonaInput } from '../shared/input/update-persona.input';
+
+@Injectable()
+export class PersonaService {
+  mongoService: MongoService<Model<PersonaDocument>>;
+
+  constructor(
+    @InjectModel('Persona')
+    private readonly personaModel: Model<PersonaDocument>,
+  ) {
+    this.mongoService = new MongoService(this.personaModel);
+  }
+
+  async createPersona(persona: PersonaInterface): Promise<PersonaDocument> {
+    return await this.mongoService.create<PersonaInterface, PersonaDocument>(
+      persona,
+    );
+  }
+
+  async updatePersona(
+    persona: UpdatePersonaInput,
+    uuid: string,
+  ): Promise<PersonaDocument> {
+    return await this.mongoService.update<UpdatePersonaInput, PersonaDocument>(
+      persona,
+      {
+        uuid,
+      },
+    );
+  }
+
+  async getPersona(condition: PersonaInput): Promise<PersonaType[]> {
```

Comment:
^ should it return array?

Reply:
yes - it call mongo find method on model with given condition:
https://mongoosejs.com/docs/api.html#model_Model.find

` find(conditions: any, projection?: any | null,
      callback?: (err: any, res: T[]) => void): DocumentQuery<T[], T> & QueryHelpers;`

Reply:
Maybe we will return first element of that finding?

Reply:
It was replaced by using findOne from mongo.